sunpy/timeseries/sources/goes.py

```diff
@@ -162,8 +124,13 @@ def _parse_file(cls, filepath):
         filepath : `str`
             The path to the file you want to parse.
         """
-        hdus = sunpy.io.read_file(filepath)
-        return cls._parse_hdus(hdus)
+        if ".fits" in Path(filepath).suffixes:
+            hdus = sunpy.io.read_file(filepath)
+            return cls._parse_hdus(hdus)
+        elif Path(filepath).suffix == ".nc":
+            return cls._parse_netcdf(filepath)
+        else:
+            return ValueError(f"{Path(filepath).name} is not supported. Only fits and netCDF (nc) can be read.")
```

Comment:
This feels like it's not robust enough to me. Inside `sunpy.io.read_file` we do a lot more clever stuff (https://github.com/sunpy/sunpy/blob/master/sunpy/io/file_tools.py#L79-L90) to detect if a file is FITS. We should write this in a way that uses that, and really, we should implement something similar for hdf5 / netcdf4 files.

I think the simplest option would just be to do the following, but we should really make our netcdf detection smarter than just the file extension at some point.
```suggestion
        if Path(filepath).suffix == ".nc":
            return cls._parse_netcdf(filepath)
        try:
            hdus = sunpy.io.read_file(filepath)
            return cls._parse_hdus(hdus)
        except UnrecognizedFileTypeError:
            return ValueError(f"{Path(filepath).name} is not supported. Only fits and netCDF (nc) can be read.")
```

The error class would need importing from here: https://github.com/sunpy/sunpy/blob/8e5a00849d8dbd355eaeb2cfc47109cdc63dd9d8/sunpy/io/file_tools.py#L218